Lib/asyncio/unix_events.py

```diff
@@ -1259,19 +1266,33 @@ def remove_child_handler(self, pid):
             return False
 
     def attach_loop(self, loop):
+        """
+        This registers the SIGCHLD signal handler.
+
+        This method can only be called from the main thread.
+        """
         # Don't save the loop but initialize itself if called first time
         # The reason to do it here is that attach_loop() is called from
         # unix policy only for the main thread.
         # Main thread is required for subscription on SIGCHLD signal
+        if loop is None or self._saved_sighandler is not None:
```

Comment:
@aeros I would think through what should happen if either `loop is None` or `self._saved_sighandler is not None`. For example, the previous implementation didn't use the `loop` argument, so there is a risk that code will stop working for people who were passing `loop=None`.